core/modules/system/system.module

```diff
@@ -2824,22 +2824,44 @@ function system_get_info($type, $name = NULL) {
  *   An associative array of module information.
  */
 function _system_rebuild_module_data() {
+  require_once BACKDROP_ROOT . '/core/includes/install.inc';
+
   // Find modules
   $modules = backdrop_system_listing('/^' . BACKDROP_PHP_FUNCTION_PATTERN . '\.module$/', 'modules', 'name', 0);
 
-  // Include the installation profile in modules that are loaded.
+  // Include the installation profile(s) in modules that are loaded.
+  // This will include the current profile, then loop through any base profiles
+  // as well.
   $profile = backdrop_get_profile();
-  $profile_uri = 'profiles/' . $profile . '/' . $profile . '.profile';
-  if (file_exists(BACKDROP_ROOT . '/core/profiles/' . $profile . '/' . $profile . '.profile')) {
-    $profile_uri = 'core/' . $profile_uri;
+  $weight = 1000;
+  do {
+    $profile_uri = 'profiles/' . $profile . '/' . $profile . '.profile';
+    if (file_exists(BACKDROP_ROOT . '/core/' . $profile_uri)) {
+      $profile_uri = 'core/' . $profile_uri;
+    }
+    $modules[$profile] = new stdClass();
+    $modules[$profile]->name = $profile;
+    $modules[$profile]->uri = $profile_uri;
+    $modules[$profile]->filename = $profile . '.profile';
+
+    // Installation profile hooks are always executed last.
+    $modules[$profile]->weight = $weight;
+    // Execute base profile hooks before sub-profile hooks.
+    $weight--;
+
+    // Get base profile, if any.
+    $profile_info = install_profile_info($profile);
+    if (isset($profile_info['base profile'])) {
+      $profile = $profile_info['base profile'];
+      $has_base = TRUE;
+    }
+    else {
+      // Set $profile back to the current one for use later in this function.
```

Comment:
```suggestion
      // Set $profile back to the current one, for use later in this function.
```

Reply:
I disagree with the use of a comma here.

Reply:
OK, consider this (when reading):

> Set $profile back to the current one, for use later in this function.

vs:

> Set $profile back to the current one for use, later in this function.

...in the last version, the person is actually reading/understanding this:

> Later in this function, set $profile back to the current one for use.

Reply:
I don't think we need a comma at all. I think it reads fine as is.